### ISSUE TITLE
Issue with NAs in first column of metadata when aggregating genomic scores in running windows

### ISSUE THREAD

**Issue by Rmulet** [ivanek/Gviz#42]
Hi,

I encountered an issue that may be related to https://github.com/ivanek/Gviz/issues/18

When using running windows to smooth the data, the presence of NAs in the **first column** of values seems to somehow influence the calculation of the average. I noticed this because every time I ordered the columns differently, the line also looked different -- even though the data remained the same.

The most extreme example occurs when the first column contains multiple NAs in straight succession: the result is a "flat line" that covers the whole region containing NAs. This is not the case if that same column is in any other position except for the first. It doesn't happen with non-running windows either.

I include here a reproducible example:

```
df = data.frame(chr="chr1",start=1:100,end=1:100,value1=c(rep(NA,90),rnorm(10)),value2=rnorm(100),value3=rnorm(100))
gr = makeGRangesFromDataFrame(df,keep.extra.columns = T)`
ref_track <- GenomeAxisTrack(add35=TRUE,add53=TRUE,littleTicks=T,cex=1.2)
dTrack <- DataTrack(gr, type=c('a'),missingAsZero=F,windowSize = 10,window=-1, aggregation="mean", na.rm=T)
plotTracks(list(ref_track,dTrack),from=1,to=100)
```

Here the first column contains 90 NAs followed by 10 random values, which results in a flat line until position 84.
![image](https://user-images.githubusercontent.com/13835031/107862373-d7154f80-6e4c-11eb-8a4f-967e5f212132.png)

But if I reorder the columns in such a way that the NAs are in the second or third columns, the average is calculated normally. Besides, putting the NAs in the second or the third columns does not seem to have an effect (as one would expect):

``` 
df2 = df[,c("chr","start","end","value2","value1","value3")]
gr2 = makeGRangesFromDataFrame(df2,keep.extra.columns = T)
dTrack <- DataTrack(gr2, type=c('a'),missingAsZero=F,windowSize = 10,window=-1, aggregation="mean", na.rm=T)
plotTracks(list(ref_track,dTrack),from=1,to=100)
```
![image](https://user-images.githubusercontent.com/13835031/107862378-de3c5d80-6e4c-11eb-8bc0-af19fabf814f.png)

```
df3 = df[,c("chr","start","end","value3","value1","value2")]
gr3 = makeGRangesFromDataFrame(df3,keep.extra.columns = T)
dTrack <- DataTrack(gr3, type=c('a'),missingAsZero=F,windowSize = 10,window=-1, aggregation="mean", na.rm=T)
plotTracks(list(ref_track,dTrack),from=1,to=100)
```

![image](https://user-images.githubusercontent.com/13835031/107862417-062bc100-6e4d-11eb-81cf-aedcf5cc126a.png)

Is there anything I'm doing wrong or is it a malfunction? I don't understand the reason for this behaviour, and I haven't been able to fix it by changing any of the parameters. Setting `missingAsZero` to `True` is not a viable option because the missing data are not 0...

**Comment by ivanek** [comment]
Hi, I will have a look into it in the first week of March.

**Comment by ivanek** [comment]
Hi @Rmulet, Thanks for reporting it. It was indeed a bug. I fix it devel (>1.35.5) and release (>1.34.1).
